mongo/mongo.go

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright 2022 SkyAPM org
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package mongo
+
+import (
+	"context"
+
+	"github.com/SkyAPM/go2sky"
+	"go.mongodb.org/mongo-driver/event"
+	agentv3 "skywalking.apache.org/repo/goapi/collect/language/agent/v3"
+)
+
+const (
+	// ComponentMongo ComponentID.
+	ComponentMongo int32 = 42
+
+	// ComponentMongoDB db.type.
+	ComponentMongoDB string = "MongoDB"
+
+	// Peer peer.
+	Peer string = "mongo:27017"
+)
+
+// Middleware mongo monitor.
+func Middleware(tracer *go2sky.Tracer) *event.CommandMonitor {
+	spanMap := make(map[int64]go2sky.Span)
+	apmMonitor := &event.CommandMonitor{
+		Started: func(ctx context.Context, evt *event.CommandStartedEvent) {
+			span, _, err := tracer.CreateLocalSpan(ctx,
+				go2sky.WithSpanType(go2sky.SpanTypeEntry),
+				go2sky.WithOperationName(GetOpName(evt.CommandName)),
+			)
+			if err != nil {
+				return
+			}
+			span.SetPeer(Peer)
+			span.SetComponent(ComponentMongo)
+			span.SetSpanLayer(agentv3.SpanLayer_Database)
+			span.Tag(go2sky.TagDBType, ComponentMongoDB)
+			// span.Tag(go2sky.TagDBStatement, evt.Command.String())
```

Comment:
What happens for this statement tag?

Reply:
Some message executing the statement will be printed, but it will contain some indeterminate data.

eg. 
`{"create": "users","lsid": {"id": {"$binary":{"base64":"vtes1313Q1OnEMNazIsA1g==","subType":"04"}}},"$clusterTime": {"clusterTime": {"$timestamp":{"t":"1668405576","i":"1"}},"signature": {"hash": {"$binary":{"base64":"yFFWPXo66l1X7B91uKwSuZ3CMA8=","subType":"00"}},"keyId": {"$numberLong":"7122666363734196228"}}},"$db": “database"} `

Reply:
Yes, tracing is for the internal messages. Could you add some configurations to disable or enable this in the tag?

Reply:
Read this.

https://github.com/apache/skywalking/blob/0520ac9c72f737143e0b830305fe94266db0e584/oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/parser/listener/vservice/VirtualDatabaseProcessor.java#L79

If the statement(tag key = `db.statement` existing, the backend would be able to sample the slow commands accordingly to thresholds settings.